### PR TITLE
docs(keyring) add recovery key for 3.0

### PR DESCRIPTION
**Pull request by mayocream**
### Summary

Add recovery key API for Gateway EE 3.0.

FT-3020

### CODE REVIEW

**Comment by mayocream** [comment]
> is this part of the 3.0 release? If so, we have a Gateway3.0 branch: https://github.com/Kong/docs.konghq.com/tree/release/gateway-3.0
> 
> And the 3.0 docs use our single-sourcing plugin so they exist within the `src` directory: https://github.com/Kong/docs.konghq.com/tree/release/gateway-3.0/src/gateway

Thanks! I've changed the base branch to `release/gateway-3.0`.

File: src/gateway/plan-and-deploy/security/db-encryption.md [hunk 1]
```diff
@@ -109,6 +109,29 @@ The exported keyring should be stored in a safe location for disaster recovery
 purposes. It is not designed to be modified or decrypted before being used during
 a disaster recovery process.
 
+### Restore the Keyring
+
+As the Keyring material is encrypted with a randomly-generated symmetric key, and only stored in memory,
+the Keyring could be lost or corrupted if the Kong process is restarted.
+Either way, the Keyring can be restored by importing the exported keyring.
+
+A recommended procedure is to use the Recovery mode, you should set the `keyring_recovery_public_key`
```
Comment (Guaris) [inline src/gateway/plan-and-deploy/security/db-encryption.md:118]:
```suggestion
A recommended method for recovering a keyring is to use the Recovery mode, you should set the `keyring_recovery_public_key`
```

File: src/gateway/plan-and-deploy/security/db-encryption.md [hunk 2]
```diff
@@ -109,6 +109,29 @@ The exported keyring should be stored in a safe location for disaster recovery
...
+Kong configuration value to the public key of the recovery keypair.
+(You can also follow the Generate a Management RSA Key Pair guide to generate a keypair for this purpose.)
```
Comment (Guaris) [inline src/gateway/plan-and-deploy/security/db-encryption.md:120]: Can we link the doc here?

File: src/gateway/plan-and-deploy/security/db-encryption.md [hunk 3]
```diff
@@ -109,6 +109,29 @@ The exported keyring should be stored in a safe location for disaster recovery
...
+
+The Keyring material is encrypted with the public RSA key defined via the `keyring_recovery_public_key`
```
Comment (Guaris) [inline src/gateway/plan-and-deploy/security/db-encryption.md:122]:
```suggestion
The keyring material is then encrypted with the public RSA key defined via the `keyring_recovery_public_key`
```

File: src/gateway/plan-and-deploy/security/db-encryption.md [hunk 4]
```diff
@@ -109,6 +109,29 @@ The exported keyring should be stored in a safe location for disaster recovery
...
+Kong configuration value in the database, the recovery keypair then is used to decrypt the Keyring material in the database.
```
Comment (Guaris) [inline src/gateway/plan-and-deploy/security/db-encryption.md:123]:
```suggestion
Kong configuration value in the database. You can use this recovery key to decrypt the Keyring material in the database.
```

File: src/gateway/plan-and-deploy/security/db-encryption.md [hunk 5]
```diff
@@ -109,6 +109,29 @@ The exported keyring should be stored in a safe location for disaster recovery
...
 a disaster recovery process.
 
+### Restore the Keyring
+
+As the Keyring material is encrypted with a randomly-generated symmetric key, and only stored in memory,
+the Keyring could be lost or corrupted if the Kong process is restarted.
+Either way, the Keyring can be restored by importing the exported keyring.
+
+A recommended procedure is to use the Recovery mode, you should set the `keyring_recovery_public_key`
+Kong configuration value to the public key of the recovery keypair.
```
Comment (Guaris) [inline src/gateway/plan-and-deploy/security/db-encryption.md:119]:
```suggestion
Kong configuration value to the public key of the recovery key pair.
```

File: src/gateway/admin-api/db-encryption.md [hunk 6]
```diff
@@ -112,6 +112,40 @@ HTTP 201 Created
 HTTP 201 Created
 ```
 
+## Recovery Keyring from Database
+
+*This endpoint is only available with the `cluster` keyring strategy.*
```
Comment (Guaris) [inline src/gateway/admin-api/db-encryption.md:117]:
```suggestion
This endpoint is only available with the `cluster` keyring strategy.
```

File: src/gateway/admin-api/db-encryption.md [hunk 7]
```diff
@@ -112,6 +112,40 @@ HTTP 201 Created
...
+
+*The endpoint requires that the `keyring_recovery_public_key` Kong configuration value is defined.*
```
Comment (Guaris) [inline src/gateway/admin-api/db-encryption.md:119]:
```suggestion
The endpoint requires that the `keyring_recovery_public_key` Kong configuration value is defined.
```

**Review by Guaris — CHANGES_REQUESTED** [review]

File: src/gateway/admin-api/db-encryption.md (hunk 6 above)
Comment (mayocream) [reply src/gateway/admin-api/db-encryption.md:117]: This was to be aligned with the original text https://github.com/Kong/docs.konghq.com/blob/bcafb595f4b59b4ce951b1d92325a200b3809769/src/gateway/admin-api/db-encryption.md?plain=1#L46-L50

File: src/gateway/admin-api/db-encryption.md (hunk 7 above)
Comment (mayocream) [reply src/gateway/admin-api/db-encryption.md:119]: ditto

**Pull request merged**